tests/test_datastore.py

```diff
@@ -0,0 +1,224 @@
+from base import TestBase, unittest
+from mock import Mock, sentinel, patch, MagicMock
+from threading import Lock
+import numbers
+
+from virtwho.datastore import Datastore
+
+
+class ArbitraryTestItem(object):
+    def __init__(self, x, y, z):
+        self.x = x
+        self.y = y
+        self.z = z
+
+class TestDatastore(TestBase):
+
+    def mock_copy(self, item):
+        # Return the item with some predictable modification
+        # Used to prove that the result of copy method is used in the right
+        # places
+        return item, item
+
+    def assert_equivalent(self, item1, item2):
```

Comment:
Khowell had a good idea that I think I'll go ahead and implement for this PR. That being to rely on the objects being put into the store having a 'copy' or 'clone' method that will return a copy of the object for storage.

Alternately, we could have the objects to be put in the store implement an interface that requires a serialize and deserialize method. We could serialize the objects on the way in to the store and deserialize them on the way out.

In either of the above case we can test that the implemented method on the objects being copied, serialized or deserialized produce suitable copies. Then all we need to see is that the datastore uses the right method on the way in and out.